### PR TITLE
Add security scanning tool (CodeQL)

### PR DESCRIPTION
**Pull request by Banging12**

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
+        # Learn more about CodeQL language support at https://git.io/codeql-language-support
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
```
Comment (StephenFluin) [inline .github/workflows/codeql.yml:45]: This should be a version, not a tag

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        languages: ${{ matrix.language }}
+        # If you wish to specify custom queries, you can do so here or in a config file.
+        # By default, queries listed here will override any specified in a config file.
+        # Prefix the list here with "+" to use these queries and those in the config file.
+        # queries: ./path/to/local/query, your-org/your-repo/queries@main
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v1
```
Comment (StephenFluin) [inline .github/workflows/codeql.yml:56]: This should be a version, not a tag

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (aelmanaa) [reply .github/workflows/codeql.yml]: check new commit. _codeql-analysis.yml_ generated by [Github security](https://github.com/smartcontractkit/documentation/new/main?filename=.github%2Fworkflows%2Fcodeql-analysis.yml&workflow_template=codeql&code_scanning)

File: .github/workflows/codeql.yml (hunk 2 above)
Comment (aelmanaa) [reply .github/workflows/codeql.yml]: check new commit. _codeql-analysis.yml_ generated by [Github security](https://github.com/smartcontractkit/documentation/new/main?filename=.github%2Fworkflows%2Fcodeql-analysis.yml&workflow_template=codeql&code_scanning)

**Comment by dwightjl** [comment]
Thank you for making this PR. While we appreciate the interest, PRs should have clear description and documentation of the problem they are solving. Closing this PR for now.

**Pull request closed without merging**